src/element/textWysiwyg.tsx

```diff
@@ -68,7 +68,10 @@ export const textWysiwyg = ({
       const lines = updatedElement.text.replace(/\r\n?/g, "\n").split("\n");
       const lineHeight = updatedElement.height / lines.length;
       const maxWidth =
-        (appState.offsetLeft + appState.width - viewportX - 8) /
+        (appState.width -
+          8 -
+          ((appState.scrollX + updatedElement.x) * appState.zoom.value +
```

Comment:
```suggestion
          ((appState.offsetLeft + updatedElement.x) * appState.zoom.value +
```

It shouldn't use `scrollX` because we're in viewport coords, not in scene coords.

Using `scrollX` causes this bug:

![wysiwyg_width](https://user-images.githubusercontent.com/5153846/139879429-99fdcfd9-6aa9-46c4-ae63-d201bf54fa6d.gif)

Reply:
What things does it break?

Reply:
`maxWidth` of the wysiwyg (when zoomed). I've been trying to figure it out and it led me to a rabbit of hole of trying to get rid of the `zoom.translation` thing — I think it's the wrong way to do zooming. We should just update `scrollX/scrollY` when zooming on a cursor.

Reply:
That I understand. This is why I change the original to `appState.scrollX + updatedElement.x` because `appState.offsetLeft + viewportX` leads to the wrong `maxWidth`.  

What does using `appState.scrollX + updatedElement.x` here break, as apparently that results in the right `maxwidth` value.

Reply:
> What does using `appState.scrollX + updatedElement.x` here break? As apparently that results in the right `maxwidth` value.

Apparently not — see gif above. I'm surprised it works at all since it's in the wrong coord system (scene vs viewport).

Reply:
I think the reason it works is it is not positioning the text, merely calculating available width. `(appState.scrollX + updatedElement.x) * appState.zoom.value` will first calculate the viewport width, then convert it to scene width by applying the zoom value.

As I mention it [here](https://github.com/excalidraw/excalidraw/issues/4098#issuecomment-951280727): comparing `viewportCoordsToSceneCoords` to `sceneCoordsToViewportCoords` the math does not seem to work out.  The two should transform into each other... but they don't. Since everything else works in Excalidraw, and the sceneCoordsToViewportCoords and vice versa are at play in other places as well... I have a hard time believing that this part of the code is wrong... though I still have my doubts.